src/js/components/List/index.d.ts

```diff
@@ -54,9 +58,9 @@ export interface ListProps<ListItemType> {
   };
   margin?: MarginType;
   onActive?: (index: number) => void;
-  onClickItem?:
-    | ((event: React.MouseEvent) => void)
-    | ((event: { item?: ListItemType; index?: number }) => void);
+  onClickItem?: (
+    event: React.MouseEvent & { item: ListItemType; index: number },
+  ) => void;
```

Comment:
Based on a read-through of the code, the correct type is a function with the intersection of the event and what it gets modified with as the single argument